File: Keepr/Models/Vault.cs
namespace Keepr.Models;
public class Vault : RepoItem<int>
{
  public string CreatorId { get; set; }
  public string Name { get; set; }
  public string Description { get; set; }
  public string Img { get; set; }
  public bool? IsPrivate { get; set; }
  public Profile Creator { get; set; }
}

File: Keepr.client/src/models/Vault.js
import { Profile } from "./Account.js"

export class Vault {
  constructor(data) {
    this.id = data.id
    this.name = data.name
    this.creatorId = data.creatorId
    this.description = data.description
    this.img = data.img
    this.isPrivate = data.isPrivate
    this.creator = new Profile(data.creator)
  }
}